azure/scope/managedmachinepool.go

```diff
@@ -219,6 +219,16 @@ func (s *ManagedMachinePoolScope) DeleteLongRunningOperationState(name, service
 	futures.Delete(s.ControlPlane, name, service)
 }
 
+// UpdateMachinePoolReplicas updates capi machinepool replica count.
+func (s *ManagedMachinePoolScope) UpdateMachinePoolReplicas(replicas *int32) error {
+	patch := client.MergeFrom(s.MachinePool.DeepCopy())
+	s.MachinePool.Spec.Replicas = replicas
+	if err := s.Client.Patch(context.Background(), s.MachinePool, patch); err != nil {
```

Comment:
We should use `patchHelper.Patch()` here

Reply:
ack